### PR TITLE
Add rhel in kubermatic

### LINKED ISSUE #5206: Adding RHEL as an os option in Kubermatic
**User Story**
Since Red Hat Enterprise Linux(rhel) is now added in all cloud provider which supports rhel, the next step would be to integrate it in Kubermatic.

**Acceptance criteria**
- [ ] RHEL is available as an os option for AWS 
- [ ] RHEL is available as an os option for Azure
- [ ] RHEL is available as an os option for GCE
- [ ] RHEL is available as an os option for Kubevirt
- [ ] RHEL is available as an os option for Openstack
- [ ] RHEL is available as an os option for vSphere

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
Adding rhel as an OS option in Kubermatic, in addition to fixing the dynamic user ssh keys syncing. Previously, we hard coded the users which would exist in the home dir based on the os, where in this PR we simply check which are the users that exist in the home dir and copy the user ssh keys to those `.ssh` folder.

**Which issue(s) this PR fixes** 
Fixes kubermatic/kubermatic#5206

**Special notes for your reviewer**:
RHEL is available as an os option for AWS
RHEL is available as an os option for Azure
RHEL is available as an os option for GCE
RHEL is available as an os option for Kubevirt
RHEL is available as an os option for Openstack
RHEL is available as an os option for vSphere

**Documentation**:
The `MachineDeployment` specs are: 
RHEL_SUBSCRIPTION_MANAGER_USER string 
RHEL_SUBSCRIPTION_MANAGER_PASSWORD string
REDHAT_SUBSCRIPTIONS_OFFLINE_TOKEN string

**Does this PR introduce a user-facing change?**:
Yes 

```release-note
None
```

### CODE REVIEW

File: api/cmd/conformance-tests/main.go [hunk 1]
```diff
@@ -494,6 +497,11 @@ func getScenarios(opts Opts, log *zap.SugaredLogger) []testScenario {
 				filteredScenarios = append(filteredScenarios, scenario)
 			}
 		}
+		if osspec.Rhel != nil {
+			if !opts.excludeSelector.Distributions[providerconfig.OperatingSystemRHEL] {
+				filteredScenarios = append(filteredScenarios, scenario)
+			}
+		}
```
Comment (xrstf) [inline api/cmd/conformance-tests/main.go:504]: In the future, this code can really use some `&&` for all these if statements.

File: api/cmd/user-ssh-keys-agent/main.go [hunk 2]
```diff
@@ -138,3 +144,21 @@ func createFileIfNotExist(path string, uid, gid int) error {
 
 	return nil
 }
+
+func availableHomeUsers() ([]string, error) {
+	files, err := ioutil.ReadDir("/home")
```
Comment (xrstf) [inline api/cmd/user-ssh-keys-agent/main.go:149]: I just want to mention it once, but not all users have a home directory and not all home directories are in `/home`.

File: api/pkg/machine/convert.go [hunk 3]
```diff
@@ -69,6 +71,18 @@ func GetAPIV1OperatingSystemSpec(machineSpec clusterv1alpha1.MachineSpec) (*apiv
 		operatingSystemSpec.SLES = &apiv1.SLESSpec{
 			DistUpgradeOnBoot: config.DistUpgradeOnBoot,
 		}
+
+	case providerconfig.OperatingSystemRHEL:
+		config := &rhel.Config{}
+		if err := json.Unmarshal(decodedProviderSpec.OperatingSystemSpec.Raw, &config); err != nil {
+			return nil, fmt.Errorf("failed to parse sles config: %v", err)
```
Comment (xrstf) [inline api/pkg/machine/convert.go]: Actual change request: `RHEL` instead of `sles`.

File: api/cmd/conformance-tests/main.go (hunk 1 above)
Comment (moadqassem) [reply api/cmd/conformance-tests/main.go:504]: Just following what is already there ;)

File: api/cmd/user-ssh-keys-agent/main.go (hunk 2 above)
Comment (moadqassem) [reply api/cmd/user-ssh-keys-agent/main.go:149]: Well, even in the previous version of this agent, we were relying on what's already in the `/home` dir. This solution was only to dynamically do what's already done.

**Comment by zreigz** [comment]
looks really good, thanks for your work :+1:

**Pull request merged**